File: src/main/java/net/Chidoziealways/everythingjapanese/entity/client/chair/ChairRenderer.java
package net.Chidoziealways.everythingjapanese.entity.client.chair;

import net.Chidoziealways.everythingjapanese.entity.custom.ChairEntity;
import net.minecraft.client.renderer.culling.Frustum;
import net.minecraft.client.renderer.entity.EntityRenderer;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.resources.ResourceLocation;

public class ChairRenderer extends EntityRenderer<ChairEntity, ChairRenderState> {
    public ChairRenderer(EntityRendererProvider.Context pContext) {
        super(pContext);
    }

    @Override
    public boolean shouldRender(ChairEntity pLivingEntity, Frustum pCamera, double pCamX, double pCamY, double pCamZ) {
        return true;
    }

    @Override
    public ChairRenderState createRenderState() {
        return new ChairRenderState();
    }

    @Override
    public void extractRenderState(ChairEntity pEntity, ChairRenderState pReusedState, float pPartialTick) {
        pReusedState.setEntity(pEntity);
        super.extractRenderState(pEntity, pReusedState, pPartialTick);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/entity/client/chair/ChairRenderer.kt
package net.Chidoziealways.everythingjapanese.entity.client.chair

import net.Chidoziealways.everythingjapanese.entity.custom.ChairEntity
import net.minecraft.client.renderer.culling.Frustum
import net.minecraft.client.renderer.entity.EntityRenderer
import net.minecraft.client.renderer.entity.EntityRendererProvider

class ChairRenderer(pContext: EntityRendererProvider.Context) :
    EntityRenderer<ChairEntity, ChairRenderState>(pContext) {
    override fun shouldRender(
        pLivingEntity: ChairEntity,
        pCamera: Frustum,
        pCamX: Double,
        pCamY: Double,
        pCamZ: Double
    ): Boolean {
        return true
    }

    override fun createRenderState(): ChairRenderState {
        return ChairRenderState()
    }

    override fun extractRenderState(pEntity: ChairEntity, pReusedState: ChairRenderState, pPartialTick: Float) {
        pReusedState.entity = pEntity
        super.extractRenderState(pEntity, pReusedState, pPartialTick)
    }
}
